### PR TITLE
fix: trade token2022 solana tokens

### LINKED ISSUE #8357: Can't trade to solana Token2022 with jupiter
### Overview

Spotted by @MBMaria in  https://discord.com/channels/554694662431178782/1316422691191918754/1316916862893686836

### References and additional details

Seems like instruction build is wrong while trying to trade to token2022 type with Jupiter

### Acceptance Criteria

- Being able to trade both to same account and to external account to token2022 tokens

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by NeOMakinG**
## Description

Solana Token 2022 were broken with Jupiter (also probably when sending)

It was because getAccount was never finding the account as it was using only the SPL Program instead of Token2022

## Issue (if applicable)

closes #8357

## Risk

Low

> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

## Testing
- Try trading to a token2022 token you don't own (CWIF is a good example)
- Try trading to a token2022 token to an external address that never had CWIF
- Do the same but for accounts that already has some CWIF (already existing ATA)
- Also try to send CWIF to another account that already has and never had CWIF
- Quick verify that you can trade other SPL tokens too

### Engineering
n/a

### Operations
n/a
- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

## Screenshots (if applicable)
wuf to cwif that never existed
https://solscan.io/tx/2eEp89f3GkFeat8FTNSsfK16SYsLf8rSwXqBBtyG8z1eRRTMhMiYJqnDUv3qKSs2LtzHDpSt4i1uQzTBXFeGxcE3

Maneki to wuf never existed
https://explorer.solana.com/tx/3euT2acibe58jF8T9gQrrJzxn5usMHnHCNqsatFx91tynt3asnPeV1anBPe2RFqskdG5kan93LFR9bLjqnxmYCrq

Maneki to cwif never existed
https://explorer.solana.com/tx/2w8bMs7vKcu2WZGLjyMMvWxsRhN4SpKGbJw4sDxRRfnhaeHW5yN5fzDmA4iNjUj8fL8zgE5nEKMhUfDbwc2zD8n5

Maneki to cwif already existed
https://explorer.solana.com/tx/35fcXK63JU8ZpPSQPCUw4KkUeQBDKUvESr28xTd6CMKzxwo6SUwKLjSig4kVFroxGv5QCtpCaefsYGpHmky3YjVQ

And a few more txs, trust me I'm losing my sanity testing this with all the edge cases 😆

### CODE REVIEW

**Comment by NeOMakinG** [comment]
> Tested locally, LGTM minus Token2022 sends to address without ATA for Token2022 token
> 
> * Try trading to a token2022 token you don't own (CWIF is a good example) ✅
> 
> https://jam.dev/c/121512f0-207e-4a61-b6d8-2951172c9900
> 
> * Try trading to a token2022 token to an external address that never had CWIF ✅
> * Do the same but for accounts that already has some CWIF (already existing ATA) ✅
>   
>   * Token2022 ATA -> Token2022 no ATA is happy ✅  https://jam.dev/c/3acef718-89bf-44cb-b93a-3b2370180b99
>   * Token2022 ATE -> Token2022 ATA is happy ✅ https://jam.dev/c/6c9c9397-4aaa-481a-b479-897b8fec0afa
> * Also try to send CWIF to another account that never had CWIF 🚫
> 
> https://explorer.solana.com/tx/ao3NFANbsKtAinVzFrdAdHiYsBJeu8hAUXBWWPsnprZcfP9Gpw19LJ4KY3HV2UgKgkuxFcHg5s1QNgBsfyw44dQ https://jam.dev/c/30e737ab-f0a7-4aa3-91c4-b00d2ebc509b
> 
> * Quick verify that you can trade other SPL tokens too ✅
> 
> https://jam.dev/c/82745be2-e27c-45ab-bab4-0f306724d645

gm @gomesalexandre 
https://solscan.io/tx/5wtMgQ2Z8UYHLShfDBAP3x1P5hkaBcwdVtH46UXcA2BUwT6sJYqb4wVU3abfo3hZrz5DEWvyb2SZmdgAM4e6Fpv9

It should work now, it was not related but I fixed it anyway, we need this working!

File: packages/swapper/src/swappers/JupiterSwapper/utils/helpers.ts [hunk 1]
```diff
@@ -253,25 +260,29 @@ export const createSwapInstructions = async ({
...
-      buyAssetReferralPubKey,
-      sellAssetReferralPubKey,
-      programId: new PublicKey(JUPITER_AFFILIATE_CONTRACT_ADDRESS),
-      instructionData,
-      buyTokenId: buyAssetAddress,
-      sellTokenId: sellAssetAddress,
-      connection: adapter.getConnection(),
-    })
+  const { instruction: feeAccountInstruction, tokenAccount } = !isBuyAssetToken2022
+    ? await getFeeTokenAccountAndInstruction({
```
[the remark this answers is not in the document]
Comment (NeOMakinG) [reply packages/swapper/src/swappers/JupiterSwapper/utils/helpers.ts]: What a catch, it should actually be a blocking, fixed in [a784a1b](https://github.com/shapeshift/web/pull/8365/commits/a784a1ba34bda2a8736c523a2ecb70bf022c21b8)

I think it's legacy from my rubber duck phase while attempting to fix those token 2022 edge cases, it's already handled under the hood by getFeeTokenAccountAndInstruction, if none of the assets are SPL Tokens, then it returns undefined

Also I noticed a bug with referral account if you swap to token2022 that I fixed in [9931ddf](https://github.com/shapeshift/web/pull/8365/commits/9931ddf7ecd7fec9545157566fc829e16b4bbfd2)

Unfortunately this bug is forcing us to use `useSharedAccounts` (which is what Jupiter uses under the hood), I feel like it's less safe, but in the end it might be better for the user anyway because fee amounts are way less than before

https://solscan.io/tx/3vScHkxZwMR2ofSCkuHnHVjCpEsfvdR9d8iseyAGT6txQnm8sBKjSxMsBxaToAkk1us71SdLKK3wRwBVAZ52ArCv

**Review by gomesalexandre — APPROVED** [review]
Reested locally:

- Try trading to a token2022 token you don't own (CWIF is a good example) ✅ 

https://jam.dev/c/6398ef49-6eaf-4482-8849-6cc2b8529d10

- Try trading to a token2022 token to an external address that never had CWIF ✅ 

https://jam.dev/c/9bd4a3d2-e486-4989-bb27-72d97a0eecdc

- Do the same but for accounts that already has some CWIF (already existing ATA) ✅ 

  - Token2022 ATA -> Token2022 no ATA is happy ✅  https://jam.dev/c/9bd4a3d2-e486-4989-bb27-72d97a0eecdc
  - Token2022 ATE -> Token2022 ATA is happy ✅ https://jam.dev/c/c82b936a-3a99-4a80-9c38-d98c38c4d97c
  
- Also try to send CWIF to another account that never had CWIF ✅  

https://jam.dev/c/bd74d9ef-3203-419e-916a-48f527192c50

- Quick verify that you can trade other SPL tokens too ✅ 

https://jam.dev/c/a440e5e1-d066-42c6-a7fc-bb021d6ab7d6
